components.go

```diff
@@ -0,0 +1,135 @@
+package discordgo
+
+import (
+	"encoding/json"
+)
+
+// ComponentType is type of component.
+type ComponentType uint
+
+// MessageComponent types.
+const (
+	ActionsRowComponent ComponentType = 1
+	ButtonComponent     ComponentType = 2
+)
+
+// MessageComponent is a base interface for all message components.
+type MessageComponent interface {
+	json.Marshaler
+	Type() ComponentType
+}
+
+// UnmarshableMessageComponent a helper for components which need to be unmarshaled (like the ones in message object).
+// Since interfaces can't be unmarshaled this exists.
+type UnmarshableMessageComponent struct {
```

Comment:
Regardless of whether this type stays exported or not, "unmarshalable" represents the term more accurately than "unmarshable", even if it is a bit clunky.

Reply:
Thanks for advise.